### ISSUE TITLE
Be prepared for 32-byte addresses

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/fe#297]
### What is wrong?

https://ethereum-magicians.org/t/increasing-address-size-from-20-to-32-bytes/5485

At some point the protocol is likely to expand addresses to 32-bytes.  There may be assumptions that are being made about addresses that could at minimum be aware that this change is coming.

### How can it be fixed

:shrug: 

Just be ready for this....  in whatever way is appropriate.

**Comment by cburgdorf** [comment]
Yep, also read about it. I think it may even end up making some things about addresses more straight forward (e.g. arrays of addresses in storage which we already pad to 32 bytes to ensure an address can't span across multiple state slots)

**Comment by axic** [comment]
A relevant example is https://ethereum-magicians.org/t/address-space-extension-with-bridge-contracts/6660.  In this we show the potential for new a ABI type, `address32`, to be used next to the current `address`.
